src/com/rcv/Tabulator.java

```diff
@@ -35,6 +35,9 @@
   // e.g. roundTallies[1] contains a map of all candidate ID -> votes for each candidate in round 1
   // this structure is computed over the course of tabulation
   private final Map<Integer, Map<String, BigDecimal>> roundTallies = new HashMap<>();
+  // precinctRoundTallies is like roundTallies, but bucketed by precinct
+  private final Map<String, Map<Integer, Map<String, BigDecimal>>> precinctRoundTallies =
```

Comment:
Looks good.  Maybe just clarify the comment: precinctRoundTallies is a map of precinct -> roundTallies for that precinct.